src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -1512,6 +1586,43 @@ private SceneDisplayer CreateEditorOrganelle()
         return node;
     }
 
+    /// <summary>
+    ///   Updates the membrane and organelle placement of the preview cell.
+    /// </summary>
+    private void UpdateCellVisualization()
+    {
+        // Don't redo the membrane when not in the preview mode to avoid possible lag spikes
+        // while editing cell structure in the structure tab
+        if (!MicrobePreviewMode || !membraneOrganellePositionsAreDirty)
+            return;
+
+        // Create a new one if not yet exist
+        if (previewMicrobe == null)
+        {
+            previewMicrobe = (Microbe)microbeScene.Instance();
+            previewMicrobe.IsForPreviewOnly = true;
+
+            // Set the initial preview cell's visibility
+            previewMicrobe.Visible = MicrobePreviewMode;
+
+            world.AddChild(previewMicrobe);
+        }
+
+        // A bit hacky but this is needed to properly visualize the edited cell completely
+        var species = (MicrobeSpecies)editedSpecies.Clone();
```

Comment:
I think this temporary species should be set in the previewMicrobe once, and then the existing instance could be re-used (read back from the Microbe for modification) to not have to clone the species many times.